Add spec for Section::ListService

chrisroos opened the pull request:
This should make it easier to make changes to this service in future.

floehopper commented on spec/services/section/list_service_spec.rb:
```diff
@@ -0,0 +1,14 @@
+  it 'returns the manual and its sections' do
+    user = double(:user)
+    sections = double(:sections)
+    manual = double(:manual, sections: sections)
```
Minor: See [my earlier comment](https://github.com/alphagov/manuals-publisher/pull/1139#discussion_r118056507) about using too many mock objects.

chrisroos commented:
Thanks for the review, @floehopper. I've updated the tests so that they use realistic objects instead of `double`s (in the same way that I change the tests in PR #1139). I've rebased on master and force pushed in preparation for merging.
